### DISCUSSION TITLE
Can Video (and similar) contents fallback into a QR code for PDF?

### DISCUSSION THREAD

**Discussion by iagobaapellaniz** [quarto-dev/quarto-cli#11269] [Feature Requests]
### Description

I think the title explains the feature by itself. Here it is an example of what I want to achieve

```
::: {#fig-cern pdf-fallback=cernqr.png}

{{< video https://www.youtube.com/embed/wo9vZccmqwc >}}

The video "CERN: The Journey of Discovery"
:::
```

Or something similar. This way who prints the document has the option of to see the video, let's say, if it has a smartphone at hand.

If that QR code could be generated from the URL itself, that would be even more useful than a simple fallback, 
```
::: {#fig-cern pdf-qr-fallback=true}

{{< video https://www.youtube.com/embed/wo9vZccmqwc >}}

The video "CERN: The Journey of Discovery"
:::
```

UPDATE: Maybe it is doable with conditional content: https://quarto.org/docs/authoring/conditional.html. I will try

UPDATE2: It was easier than I thought. This would make the trick,`\usepage{qrcode}` in the preamble, and
```
::: {#fig-cern}

\qrcode{https://www.youtube.com/embed/wo9vZccmqwc}

{{< video https://www.youtube.com/embed/wo9vZccmqwc >}}

The video "CERN: The Journey of Discovery"
:::
```
since LaTeX literal is ignored for HTML, it won't show up in the website. Whereas, for the PDF, we will have a nice QR code + Hyperlink

**Comment by mcanouil** [comment]
You could write a Lua filter to do that.

FYI, https://github.com/jmbuhr/quarto-qrcode (http://m.canouil.dev/quarto-extensions/)
